### PR TITLE
closes #1105 Make brands clickable on Product Page

### LINKED ISSUE #1105: Make brands clickable on Product Page
One of the killer things on Open Food Facts web version is being able to browse all products of a same brand by clicking on the brand field on the product page (eg: Haldiram's)
It would be great to make the brand field clickable

https://world.openfoodfacts.org/product/8904063200013/aloo-bhijia-200g-kartoffelsnack-scharf-haldirams-haldiram-s

Example with Haldiram's
https://world.openfoodfacts.org/brand/haldiram-s
https://world.openfoodfacts.org/brand/haldiram-s.json

### PR DESCRIPTION
**Pull request by Shubham2320**
## Description

- Made brands clickable on the product page.

- Created Model class brand to hold the JSON result of API Call.

- Created BrandActivity to show the list of products the brand contains in recyclerView.

## Related issues and discussion
closes #1105
 
 ## Screen-shots, if any

**Loading Screen:** 

**Products List:**

**Error Screen:**

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
Nice .

**Comment by Shubham2320** [comment]
Thanx @PrajwalM2212 :).

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java [hunk 1]
```diff
@@ -65,9 +66,13 @@
     @POST("/cgi/product_image_upload.pl")
     Call<JsonNode> saveImage(@PartMap Map<String, RequestBody> fields);
 
+    @GET("brand/{brand}.json")
+    Call<Brand> getBrand(@Path("brand") String brand);
```
Comment (jaztriumph) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: I think `byBrand` also does the same thing there is no need for this function. we can use `Search.java`  for this purpose there is no need for `Brand.java` that's almost the same.

Comment (jaztriumph) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: we can paginate the results and implement infinite scrolling same way as it is done in `searchProductByName` function, if not loading will be sluggish.

Comment (Shubham2320) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: yeah, sure I would look at it, thanx :). Also, I'm working to get other pages of the list as well.

**Comment by Shubham2320** [comment]
@Karljoones I've made the changes also added infinite recycler view and as @jaztriumph has suggested I've used Search.java model class instead of creating a new one to hold the response. Please review :).

**Comment by teolemon** [comment]
small conflict to merge

**Comment by Karljoones** [comment]
Merged, thanks @Shubham2320 !

**Pull request merged**